src/connectivity.jl

```diff
@@ -305,43 +305,52 @@ function attracting_components end
 end
 
 """
-    neighborhood(g, v, d)
+    neighborhood(g, v, d, distmx=weights(g))
+    neighborhood_with_distances(g, v, d, distmx=weights(g))
 
-Return a vector of the vertices in `g` at a geodesic distance less or equal to `d`
-from `v`.
+Return a vector of each vertex in `g` at a geodesic distance less than or equal to `d`, where distances
+may be specified by `distmx`. 
+
+For `neighborhood_with_distances`, return a tuple of each vertex with its distance from `v`.
 
 ### Optional Arguments
 - `dir=:out`: If `g` is directed, this argument specifies the edge direction
 with respect to `v` of the edges to be considered. Possible values: `:in` or `:out`.
 """
-neighborhood(g::AbstractGraph, v::Integer, d::Integer; dir=:out) = (dir == :out) ?
-    _neighborhood(g, v, d, out_neighbors) : _neighborhood(g, v, d, in_neighbors)
+neighborhood(g::AbstractGraph{T}, v::Integer, d, distmx::AbstractMatrix{U}=weights(g); dir=:out) where T<:Integer where U<:Real =
+    first.(neighborhood_with_distances(g, v, d, distmx; dir=dir))
+
+neighborhood_with_distances(g::AbstractGraph{T}, v::Integer, d, distmx::AbstractMatrix{U}=weights(g); dir=:out) where T<:Integer where U<:Real =
+    (dir == :out) ? _neighborhood(g, v, d, distmx, out_neighbors) : _neighborhood(g, v, d, distmx, in_neighbors)
 
-function _neighborhood(g::AbstractGraph{T}, v::Integer, d::Integer, neighborfn::Function) where T
-    d < 0 && return Vector{T}()
+
+function _neighborhood(g::AbstractGraph{T}, v::Integer, d::Real, distmx::AbstractMatrix{U}, neighborfn::Function) where T<:Integer where U<:Real
+    d < 0 && return Vector{Tuple{T, U}}()
```

Comment:
Nice a null result rather than `error` i like it.

Reply:
It makes sense, right? Nothing in your negative neighborhood.